assets/config.php

```diff
@@ -283,7 +283,7 @@
         // Stancl\Tenancy\Features\TelescopeTags::class,
         // Stancl\Tenancy\Features\UniversalRoutes::class,
         // Stancl\Tenancy\Features\TenantConfig::class, // https://tenancyforlaravel.com/docs/v3/features/tenant-config
-        // Stancl\Tenancy\Features\CrossDomainRedirect::class, // https://tenancyforlaravel.com/docs/v3/features/cross-domain-redirect
+        Stancl\Tenancy\Features\CrossDomainRedirect::class, // https://tenancyforlaravel.com/docs/v3/features/cross-domain-redirect
```

Comment:
Probably wouldn't change the default here — the features are *optional features*, so that implies they should all be disabled by default.

If you need to change something in the config for tests, use the `TestCase` for that.

Reply:
Don't really get your comment.

> I thought that's wht we moved to a separate TestCase thing.

And here you're saying that it's **not** the reason. That it has to be moved anyway:
> it has the same issue. SP runs before TestCase. always bootstrap feature never registers in this way. That's why I moved the whole RedirecTest suite into without tenancy.

So that seems contradictory. Either way, it's not what I'm suggesting — **I'm not talking about tests at all here, or the reason for moving anything**. I'm talking about the fact that it should remain disabled by default.

I don't think the test being separate has anything to do with whether it's enabled or disabled by default. We're using a dataset so both cases are covered. The test is separate to avoid running the service provider. I'm not suggesting that this change has anything to do with the tests, I'm simply telling you that the feature should remain optional like it was in v3.

So I don't get what you mean by your comment.

Reply:
What I meant by this:

> If you need to change something in the config for tests, use the TestCase for that.

Is that **if you made this change because of tests**, change the config that's set inside `TestCase` (open the file to see what I mean).

If you made this change for any other reason, then you can share it or revert the change.

I'm not saying that this is because of tests. I just went a step ahead and said that **if** this is due to tests, then you can set the config in the `TestSuite`.

Or is your entire point that `getEnvironmentSetUp()` won't work because the SP runs before that, so to even test the redirect in the **normal** tests we need this to be enabled in the config? I guess that is what you mean here, but your comment is pretty confusing so it only occurred to me now that this may be what you mean. Please confirm if that's what you were trying to say.

If that's the case, then you might be able to set the config using some `resolving()` hook or something like that in `setUp()` of that test. If that won't work, then maybe changing the features to be bootstrapped using `extend()` instead of `resolving()` might make `resolving()` work for changing the config.

Regarding the `extend()` / `resolving()`, I think I had another review for that. We should be intentional about whether we use `extend()` or `resolving()`, it should have a reason.

Reply:
> Or is your entire point that getEnvironmentSetUp() won't work because the SP runs before that, so to even test the redirect in the normal tests we need this to be enabled in the config? I guess that is what you mean here, but your comment is pretty confusing so it only occurred to me now that this may be what you mean. Please confirm if that's what you were trying to say.

Yes that's what I was trying to say. 

> If that won't work, then maybe changing the features to be bootstrapped using extend() instead of resolving() might make resolving() work for changing the config

`resolving` container event works but we want to always bootstrap feature. The bootstrapping feature in `resolving` means the feature is not always bootstrap. The issue attached to this PR explains that OP wants to use the feature without using `Tenancy`.

Reply:
Ah right, I remembered that we had this code in an earlier version of the PR but forgot that it was changed later because of this.

So do you not see any solutions for enabling the feature in the config, in tests? If you don't, I'll find one and finish this PR myself.

Reply:
Enabling the feature in `getPackageProviders`  works but not sure if this is clean/right approach. 

```php
    protected function getPackageProviders($app)
    {
        config()->set('tenancy.features', [CrossDomainRedirect::class]);

        return [
            TenancyServiceProvider::class,
        ];
    }
```

I'm leaving this review open. If you can find a better approach, Please finish PR.

Reply:
I think I'll add a static property for disabling parts of the Tenancy SP, and a method for running them manually at a point that suits us in tests. Will complete this tomorrow probably.

Reply:
I think we can resolve this now @stancl (https://github.com/archtechx/tenancy/compare/258582112cbfedeb39b2a2100281f15bfc020e41..06b62b85cd362cac5d3689636aded57595c10f18)